tests/network/conftest.py

```
@@ -281,11 +282,20 @@ def _verify_sriov():
                     f"has {len(sriov_workers)} SRIOV-capable worker nodes"
                 )
 
+    def _verify_ipv4():
+        if any(test.get_closest_marker("ipv4") for test in collected_tests):
+            LOGGER.info("Verifying if the cluster supports running IPV4 tests...")
+            if not ipv4_supported_cluster:
+                failure_msgs.append("IPv4 is not supported in this cluster")
+            else:
+                LOGGER.info("Validated network lane is running against an IPV4 supported cluster")
+
     _verify_multi_nic()
     _verify_dpdk()
     _verify_service_mesh()
     _verify_jumbo_frame()
     _verify_sriov()
+    _verify_ipv4()
```

Please rearrange these by order so we will fail faster for most common cases - I mean that verify_ipv4 cluster should be second to verify_multi_nic, than the jumbo and sriov and last service mesh and dpdk.

---

The order of these functions does not affect the failure time, as all functions are executed to construct the failure messages. This ensures that all failures are logged, not just the first one. The actual failure is triggered at the end by exit_pytest_execution, which is executed outside of all the function calls.

---

I see, thank you for the explanation.
Why not break on the first missing component? Let's say that we need a multi-nic cluster and the cluster we are running on is a single-nic - why continue the checks?

---

We continue the checks to provide the user with all relevant information about the cluster's state.

---

Ah, also makes sense. Thanks, you can resolve this conversation.